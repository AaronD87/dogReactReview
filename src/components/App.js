import React, { PureComponent } from 'react';
import AddDogForm from './AddDogForm';
import Dogs from './Dogs';
// import styles from './App.css';

class MyClass extends PureComponent {
  state = {
    dogs: []
  };

  addDog = dog => {
    this.setState(state => ({
      dogs: [...state.dogs, dog]
    }));
  };

  render() {
    const { dogs } = this.state;

    return (
      <div >
        <h1>Dogs</h1>
        <div >
          <AddDogForm addDog={this.addDog}/>

          <Dogs
            dogs={dogs}
          />

        </div>
      </div>
    );
  }
}

export default MyClass;

