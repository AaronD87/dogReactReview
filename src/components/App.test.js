import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('renders the App', () => {
    const dogs = [{ age: 12, name: 'Billy', weight: '70 lbs' }];
    const wrapper = shallow(<App dogs={ dogs } />);

    expect(wrapper).toMatchSnapshot();
  });
});
