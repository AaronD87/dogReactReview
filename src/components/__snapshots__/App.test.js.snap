// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the App 1`] = `
<div>
  <h1>
    Dogs
  </h1>
  <div>
    <AddDogForm
      addDog={[Function]}
    />
    <Dogs
      dogs={Array []}
    />
  </div>
</div>
`;
